### PR TITLE
Document `Uppercase`, `Lowercase`, `Capitalize` and `Uncapitalize` built-in types

### PR DESCRIPTION
**Pull request by kainiedziela**
[`Uppercase<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms every character in a string into uppercase.

```ts
type T10 = Uppercase<'hello'>;  // "HELLO"

type T11 = Uppercase<'foo' | 'bar'>;  // "FOO" | "BAR"

type T12<S extends string> = Uppercase<`aB${S}`>;
type T13 = T30<'xYz'>;  // "ABXYZ"

type T14 = Uppercase<string>;  // string
type T15 = Uppercase<any>;  // any
type T15 = Uppercase<never>;  // never
type T16 = Uppercase<42>;  // Error, type 'number' does not satisfy the constraint 'string'
```

[`Lowercase<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms every character in a string into lowercase.

```ts
type T20 = Lowercase<'HELLO'>;  // "hello"

type T21 = Lowercase<'FOO' | 'BAR'>;  // "foo" | "bar"

type T22<S extends string> = Lowercase<`aB${S}`>;
type T23 = T32<'xYz'>;  // "abxyz"

type T24 = Lowercase<string>;  // string
type T25 = Lowercase<any>;  // any
type T26 = Lowercase<never>;  // never
type T27 = Lowercase<42>;  // Error, type 'number' does not satisfy the constraint 'string'
```

[`Capitalize<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms the first character in a string into uppercase.

```ts
type T30 = Capitalize<'hello'>;  // "Hello"

type T31 = Capitalize<'foo' | 'bar'>;  // "Foo" | "Bar"

type T32<S extends string> = Capitalize<`aB${S}`>;
type T33 = T32<'xYz'>;  // "ABxYz"

type T34 = Capitalize<string>;  // string
type T35 = Capitalize<any>;  // any
type T36 = Capitalize<never>;  // never
type T37 = Capitalize<42>;  // Error, type 'number' does not satisfy the constraint 'string'
```

[`Uncapitalize<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms the first character in a string into lowercase.

```ts
type T40 = Uncapitalize<'Hello'>;  // "hello"

type T41 = Uncapitalize<'Foo' | 'Bar'>;  // "foo" | "bar"

type T42<S extends string> = Uncapitalize<`AB${S}`>;
type T43 = T30<'xYz'>;  // "aBxYz"

type T44 = Uncapitalize<string>;  // string
type T45 = Uncapitalize<any>;  // any
type T46 = Uncapitalize<never>;  // never
type T47 = Uncapitalize<42>;  // Error, type 'number' does not satisfy the constraint 'string'
```

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -632,6 +632,86 @@ There are many advanced types most users don't know about.
 	```
 	</details>
 
+    - [`Uppercase<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms every character in a string into uppercase.
+    <details>
+    <summary>
+        Example
+    </summary>
+    ```ts
+    type T10 = Uppercase<'hello'>;  // "HELLO"
```
Comment (sindresorhus) [inline readme.md:641]: Should start at `T` and then `T2` in each code block.

File: readme.md [hunk 2]
```diff
@@ -632,6 +632,86 @@ There are many advanced types most users don't know about.
...
+    type T15 = Uppercase<never>;  // never
+    type T16 = Uppercase<42>;  // Error, type 'number' does not satisfy the constraint 'string'
+    ```
+    </details>
+
+    - [`Lowercase<S extends string>`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) - Transforms every character in a string into lowercase.
+    <details>
+    <summary>
+        Example
+    </summary>
```
Comment (sindresorhus) [inline readme.md:659]: Use-tab indentation.

**Comment by sindresorhus** [comment]
The rendered output does not look correct: https://github.com/kainiedziela/type-fest/blob/document-built-in-types/readme.md#built-in-types

**Comment by kainiedziela** [comment]
Fixed

**Pull request merged**